src/Friends/ProfileTrait.php

```diff
@@ -22,8 +22,14 @@ public function getProfile()
     {
         if (empty($this->profile)) {
             $nickname = \uniqid(__FUNCTION__ . "-");
-            $this->getContainer()->add($nickname, Profile::class)
-                ->addArgument([$this->get('profile')]);
+            if (!empty($this->attributes->profile)) {
+                $this->getContainer()->add($nickname, new Profile($this->attributes->profile))
```

Comment:
Pragmatically, this works, but I have some reservations. Do you know why this works? What you are saying is:

- Create a new profile object, and construct it with $this->attributes->profile (constructor parameter)
- Add it to the container under the provided nickname
- When a new object is constructed from the container, pass it $this->get('profile') as the constructor parameter

I'm not sure which constructor is running; presumably the first, since otherwise this PR would not change the behavior of the code.

Would this also work?
```
$this->getContainer()->add($nickname, Profile::class)
                ->addArgument([$this->attributes->profile]);
```
Also, see: https://github.com/pantheon-systems/terminus/pull/2187/files#diff-77c3583e23e3435915e890a0a58123f6a176e0e50c4e3b5f81e7ee37604488d7R24

Reply:
This pointed me in the right direction. The fix was easier than expected.